src/modules/Pitcher/pitcher.py

```diff
@@ -1,51 +1,99 @@
-    blue_highlighted,
-    red_highlighted,
-)
+import torchcrepe
+from torch.cuda import OutOfMemoryError
+
+from modules.console_colors import ULTRASINGER_HEAD, blue_highlighted, red_highlighted
 from modules.Pitcher.pitched_data import PitchedData
 
+DEFAULT_MODEL: str = "full"
+SUPPORTED_MODELS: list[str] = ["tiny", "full"]
+TIMES_DECIMAL_PLACES: int = 3
+
 
-def get_pitch_with_crepe_file(filename: str, step_size: int, model_capacity: str) -> PitchedData:
+def get_pitch_with_crepe_file(
+    filename: str,
+    model: str,
+    device: str = "cpu",
+    batch_size: int = None,
+    step_size: int = 10,
+) -> PitchedData:
     """Pitch with crepe"""
-    print(
-        f"{ULTRASINGER_HEAD} Pitching with {blue_highlighted('crepe')} and model {blue_highlighted(model_capacity)} and {red_highlighted('cpu')} as worker"
-    )
-    # Todo: add GPU support by using torchcrepe
-    sample_rate, audio = wavfile.read(filename)
-
-    pitched_data = PitchedData()
-    (
-        pitched_data.times,
-        pitched_data.frequencies,
-        pitched_data.confidence,
-        activation,
-    ) = crepe.predict(
-        audio, sample_rate, model_capacity, step_size=step_size, viterbi=True
+
+    # Load audio
+    audio, sample_rate = torchcrepe.load.audio(filename)
+
+    return get_pitch_with_crepe(
+        audio, sample_rate, model, device, batch_size, step_size
     )
-    return pitched_data
 
 
-def get_pitch_with_crepe(audio, sample_rate: int, step_size: int, model_capacity: str) -> PitchedData:
+def get_pitch_with_crepe(
+    audio,
+    sample_rate: int,
+    model: str,
+    device: str = "cpu",
+    batch_size: int = None,
+    step_size: int = 10,
+) -> PitchedData:
     """Pitch with crepe"""
-    pitched_data = PitchedData()
-    (
-        pitched_data.times,
-        pitched_data.frequencies,
-        pitched_data.confidence,
-        activation,
-    ) = crepe.predict(
-        audio, sample_rate, model_capacity, step_size=step_size, viterbi=True
+
+    if model not in SUPPORTED_MODELS:
+        print(
+            f"{ULTRASINGER_HEAD} {blue_highlighted('crepe')} model {blue_highlighted(model)} is not supported, defaulting to {blue_highlighted(DEFAULT_MODEL)}"
+        )
+        model = DEFAULT_MODEL
+
+    print(
+        f"{ULTRASINGER_HEAD} Pitching using {blue_highlighted('crepe')} with model {blue_highlighted(model)} and {red_highlighted(device)} as worker"
     )
-    return pitched_data
+
+    hop_length = None
+    if step_size:
+        step_size_seconds = round(step_size / 1000, TIMES_DECIMAL_PLACES)
+        steps_per_second = 1 / step_size_seconds
+        hop_length = sample_rate // steps_per_second
+    else:
+        step_size_seconds = round(step_size / 1000, TIMES_DECIMAL_PLACES)
+        steps_per_second = 1 / step_size_seconds
+        hop_length = sample_rate // steps_per_second
+
+    # Provide a sensible frequency range for your domain (upper limit is 2006 Hz)
+    # TODO: determine appropriate range for vocals
```

Comment:
Difficult. The record is lower 0,189hz and 25000hz. But typically, it should be around ~87hz to ~1397hz. I think your values are good. Maybe start an issue or discussion? Or maybe adding a flag to open all frequencies?

Reply:
I'll start a discussion to ask for feedback regarding this. What do you think about introducing two cli arguments to leave it up to the user to specify custom bounds?

I.e.:

`--crepe_frequency_limit_lower`
`--crepe_frequency_limit_upper`

Or something like that

Reply:
I have already found out the hard way the lower bound of 50Hz is creating strange results. I ran the program on a file of me singing test and it produced this graph when the range was 50-2006 (albeit with pretty much 0 confidence, I had to allow plotting of confidences below 0.4 to produce it):
![plot](https://github.com/rakuri255/UltraSinger/assets/19321042/a23e662c-ecbd-4eb0-bdfa-60af300b7f6a)

With a range from 0-2006 this did not happen and confidence was high were appropriate:
![plot](https://github.com/rakuri255/UltraSinger/assets/19321042/7e9354d4-31d1-4467-8a8a-4b333ab233e8)

So for now I suggest we enable the entire range.

Reply:
Yes lets for now enable the full range and also the add the flags, so more people can test and hopefully give an feedback